File: exercise/java/day24/src/test/java/delivery/ToyBuilder.java
package delivery;

import com.github.javafaker.Faker;
import domain.Toy;

public class ToyBuilder {
    private final int stock;

    private ToyBuilder(int stock) {
        this.stock = stock;
    }

    public static ToyBuilder toysInStock(int stock) {
        return new ToyBuilder(stock);
    }

    public Toy build() {
        return Toy.create(Time.PROVIDER, Faker.instance().lorem().word(), stock)
                .getOrElseThrow(() -> new RuntimeException("Toy creation failed"));
    }
}

File: exercise/C#/day24/tests/Delivery.Tests/ToyBuilder.cs
using Delivery.Domain;

namespace Delivery.Tests;

public class ToyBuilder
{
    private readonly int _stock;

    private ToyBuilder(int stock) => _stock = stock;

    public static ToyBuilder ToysInStock(int stock = 1) => new(stock);

    public Toy Build() => Toy.Create(Time.Provider, Faker.Lorem.GetFirstWord(), _stock).RightUnsafe();
}